tests/BuildScriptGenerator.Common.Tests/VersionInfoTest.cs

```diff
@@ -0,0 +1,60 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using Xunit;
+using Microsoft.Oryx.Tests.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Common.Tests
+{
+    public class VersionInfoTest : IClassFixture<TestTempDirTestFixture>
+    {
+        public VersionInfoTest(TestTempDirTestFixture testFixture)
+        {
+        }
+
+        [Theory]
+        [InlineData("3.9.0b1", "3.9.0-b1")]
+        [InlineData("3.8.0-b3", "3.8.0-b3")]
+        [InlineData("12.14.0", "12.14.0")]
+        [InlineData("3.9.0b1.23", "3.9.0-b1.23")]
+        [InlineData("5.0.0-preview.2.20160.6", "5.0.0-preview.2.20160.6")]
+        public void VersionInfo_ConvertDisplayVersionToSemanticVersion(string displayVersion, string expectedVersion)
+        {
+            // Act
+            VersionInfo versionInfo = new VersionInfo(displayVersion);
```

Comment:
`var versionInfo`..similarly in other places.

Reply:
a general question, why do we use `var ` in our code base instead of specific types, I think it's more convenient for us to debug

Reply:
Code is less verbose and less things to type.